### PR TITLE
Hotfix/simplify from thread channel

### PR DESCRIPTION
**Pull request by mosquito**

### CODE REVIEW

**Review by dizballanze — CHANGES_REQUESTED** [review]

File: aiomisc/iterator_wrapper.py [hunk 1]
```diff
@@ -1,12 +1,14 @@
 import asyncio
 import inspect
-import threading
 from collections import deque
 from concurrent.futures import Executor
+from queue import Empty as QueueEmpty
+from queue import Queue
```
[the remark this answers is not in the document]
Comment (mosquito) [reply aiomisc/iterator_wrapper.py:6]: Gray decided this good

File: aiomisc/iterator_wrapper.py [hunk 2]
```diff
@@ -25,49 +27,53 @@ class ChannelClosed(RuntimeError):
...
+    def __init__(self, max_size: int):
+        self.max_size = max_size
 
-    def __notify_writers(self) -> None:
-        with self.__write_condition:
-            self.__write_condition.notify_all()
+        if max_size > 0:
+            self.queue = Queue(maxsize=max_size)
+        else:
+            self.deque = deque()
```
[the remark this answers is not in the document]
Comment (mosquito) [reply aiomisc/iterator_wrapper.py:43]: I don't want to check the instances in runtime

**Review by Alviner — APPROVED** [review]

**Review by dizballanze — CHANGES_REQUESTED** [review]

File: aiomisc/iterator_wrapper.py [hunk 3]
```diff
@@ -25,49 +28,66 @@ class ChannelClosed(RuntimeError):
...
-        self._loop = loop
-        self.__close_event = threading.Event()
-        self.__write_condition = threading.Condition()
-        self.__read_event = asyncio.Event()
-        self.__get_lock = asyncio.Lock()
 
-    def __notify_readers(self) -> None:
-        self.loop.call_soon_threadsafe(self.__read_event.set)
+    def get(self) -> Any:
+        if not len(self.queue):
```
[the remark this answers is not in the document]
Comment (mosquito) [reply aiomisc/iterator_wrapper.py:47]: Explicit is better then implicit

[the remark this answers is not in the document]
Comment (dizballanze) [reply aiomisc/iterator_wrapper.py:47]: https://peps.python.org/pep-0008/#programming-recommendations

image

**Review by alvassin — APPROVED** [review]

**Review by dizballanze — APPROVED** [review]

**Pull request merged**